File: s02e02_pet_java/java/src/Pet.java
import java.util.Scanner;

public class Pet {
    String name;
    int energy;
    int hungry;
    int clean;
    int energyMax;
    int hungryMax;
    int cleanMax;

    public Pet(String name, int energy, int hungry, int clean){
        this.name = name;
        this.energy = energy;
        this.hungry = hungry;
        this.clean = clean;
        this.energyMax = energy;
        this.hungryMax = hungry;
        this.cleanMax = clean;
    }

    public String toString() {
        return  "[" + this.name + "] " +
                "E: " + this.energy + "/" + this.energyMax + " " +
                "S: " + this.hungry + "/" + this.hungryMax + " " +
                "L: " + this.clean + "/" + this.cleanMax;
    }
}

class Controller{
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        Pet pet = new Pet("", 0, 0, 0);
        System.out.println("end, init _nome _E _S _L, show");
        while(true){
            String line = scan.nextLine();
            String[] ui = line.split(" ");
            if(ui[0].equals("end")){
                break;
            }else if(ui[0].equals("init")){
                pet = new Pet(ui[1], Integer.parseInt(ui[2]),
                                     Integer.parseInt(ui[3]),
                                     Integer.parseInt(ui[4]));
            }else if(ui[0].equals("show")){
                System.out.println(pet);
            }else{
                System.out.println("fail: comando invalido");
            }
        }
    }
}

File: s02e02_pet_java/pet.py
#!/usr/bin/env python3

class Pet:
    #construtor
    def __init__(self, name, energy, hungry, clean):
        self.name = name
        self.energy = energy
        self.hungry = hungry
        self.clean = clean
        self.energyMax = energy
        self.hungryMax = hungry
        self.cleanMax = clean

    def __str__(self):
        return "[" + self.name + "] E: " + str(self.energy) + "/" + str(self.energyMax) + " " + \
               "S: " + str(self.hungry) + "/" + str(self.hungryMax) + " " + \
               "L: " + str(self.clean) + "/" + str(self.cleanMax) + " "

pet = Pet("", 0, 0, 0)
print("end, init _nome _E _S _L, show")
while True:
    ui = input().split(" ")
    if ui[0] == "end":
        break
    elif ui[0] == "init":
        pet = Pet(ui[1], int(ui[2]), int(ui[3]), int(ui[4]))
    elif ui[0] == "show":
        print(pet)
    else:
        print("fail: comando invalido")


               